server/models/SearchQuery.js

```diff
@@ -0,0 +1,65 @@
+// @flow
+import { DataTypes, sequelize } from "../sequelize";
+
+const SearchQuery = sequelize.define(
+  "search_queries",
+  {
+    id: {
+      type: DataTypes.UUID,
+      defaultValue: DataTypes.UUIDV4,
+      primaryKey: true,
+    },
+    source: {
+      type: DataTypes.ENUM("slack", "app"),
+      allowNull: false,
+    },
+    query: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    results: {
+      type: DataTypes.NUMBER,
+      allowNull: false,
+      defaultValue: 1,
+    },
+  },
+  {
+    timestamps: true,
+    updatedAt: false,
+  }
+);
+
+SearchQuery.associate = (models) => {
+  SearchQuery.belongsTo(models.User, {
+    as: "user",
+    foreignKey: "userId",
+  });
+  SearchQuery.belongsTo(models.Team, {
+    as: "team",
+    foreignKey: "teamId",
+  });
+};
+
+SearchQuery.saveQuery = async function (searchQuery) {
```

Comment:
Should this method be a beforeCreate hook?

Reply:
I don't think this method needs to exist, `SearchQuery.create` can be called directly from the controllers.

The key thing that was misunderstood is that the `results` should record the number of search results that were returned to the user, not the number of times the user has performed the search

Reply:
Got it, I'll fix it.